lib/TPP/ConvertTppToXsmm.cpp

```diff
@@ -342,65 +342,85 @@ struct ConvertTppIdentityOp : public OpRewritePattern<tpp::IdentityOp> {
     computeBcastShapeInput(shapeOutput, shapeInput, bShapeInput);
     assert(shapeOutput.size() == bShapeInput.size());
     shapeInput = bShapeInput;
-
-    if (shapeInput[1] == 1 && shapeOutput[1] > 1) {
-      xsmm::UnaryFlags bCast = xsmm::UnaryFlags::BCAST_ROW;
-      int64_t ldi = *getLeadingDim(outputType.cast<MemRefType>(), 1);
-      return {ldi, bCast};
+    assert(shapeInput.size() == 1 || shapeInput.size() == 2);
+
+    // Handle 2d-case.
+    if (shapeInput.size() == 2) {
+      if (shapeInput[1] == 1 && shapeOutput[1] > 1) {
+        xsmm::UnaryFlags bCast = xsmm::UnaryFlags::BCAST_ROW;
+        int64_t ldi = *getLeadingDim(outputType.cast<MemRefType>(), 1);
```

Comment:
None of these calls are checking for failures. This will cause a segfault on pointer dereference.